### PR TITLE
feat(headless): started using the execute search action compatible with the new event protocol in all the insight controllers

### PR DESCRIPTION
**Pull request by mmitiche**
## [SFINT-5453](https://coveord.atlassian.net/browse/SFINT-5453)

In a previous PR the executeSearch action has been refactored to start supporting the search requests sent that follow the new event protocol: [#3678](https://github.com/coveo/ui-kit/pull/3678)
However the `legacy execute search` was still being used because no analytics parameter `next` was not provided to all the executeSearch calls made in the insight controllers.

In the PR, I added this `next` parameter, so now when `analyticsMode` will be set to `next`, all the insight controllers will be using the new execute search compatible with event protocol instead of the legacy execute search.

#### before:
```
engine.dispatch(executeSearch({legacy: logInterfaceLoad()}));
```
#### after:
```
engine.dispatch(executeSearch({legacy: logInterfaceLoad(), next: interfaceLoad()}));
```

Additionally added new insight analytics migration to make sure the search requests made with an engine working with legacy analytics  are equivalent to the search requests made with an engine working with the next analytics.

### Demo:
Screenshot 2024-03-22 at 10 50 14 AM

[SFINT-5453]: https://coveord.atlassian.net/browse/SFINT-5453?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/features/pagination/pagination-analytics-actions.ts [hunk 1]
```diff
@@ -35,3 +37,19 @@ export const logPagePrevious = (): LegacySearchAction =>
       pagerNumber: currentPageSelector(state as PaginationSection),
     })
   );
+
```
Comment (louis-bompart) [inline packages/headless/src/features/pagination/pagination-analytics-actions.ts:40]:
```suggestion
// TODO KIT-2983
```

We're gonna change'em

**Review by louis-bompart — APPROVED** [review]
noice

**Comment by louis-bompart** [comment]
DId you do some manual test to see if the events are sane with analyticsMode: next?

**Comment by mmitiche** [comment]
> DId you do some manual test to see if the events are sane with analyticsMode: next?

Yep I did 👍

**Review by SimonMilord — APPROVED** [review]
👌

**Review by erocheleau — APPROVED** [review]
Very nice! :+1:

**Pull request merged**